packages/modules/cra-template-web-viewer/template/src/App.tsx

```diff
@@ -44,121 +44,47 @@ const App: React.FC = () => {
     );
   }
 
-  const authConfig: BrowserAuthorizationClientConfiguration = {
-    scope: process.env.IMJS_AUTH_CLIENT_SCOPES ?? "",
-    clientId: process.env.IMJS_AUTH_CLIENT_CLIENT_ID ?? "",
-    redirectUri: process.env.IMJS_AUTH_CLIENT_REDIRECT_URI ?? "",
-    postSignoutRedirectUri: process.env.IMJS_AUTH_CLIENT_LOGOUT_URI,
-    responseType: "code",
-    authority: process.env.IMJS_AUTH_AUTHORITY,
-  };
-
-  useEffect(() => {
-    if (isAuthorized) {
-      const urlParams = new URLSearchParams(window.location.search);
-      if (urlParams.has("contextId")) {
-        setContextId(urlParams.get("contextId") as string);
-      } else {
-        if (!process.env.IMJS_CONTEXT_ID) {
-          throw new Error(
-            "Please add a valid context ID in the .env file and restart the application or add it to the contextId query parameter in the url and refresh the page. See the README for more information."
-          );
-        }
-      }
-
-      if (urlParams.has("iModelId")) {
-        setIModelId(urlParams.get("iModelId") as string);
-      } else {
-        if (!process.env.IMJS_IMODEL_ID) {
-          throw new Error(
-            "Please add a valid iModel ID in the .env file and restart the application or add it to the iModelId query parameter in the url and refresh the page. See the README for more information."
-          );
-        }
-      }
-    }
-  }, [isAuthorized]);
-
-  useEffect(() => {
-    if (contextId && iModelId && isAuthorized) {
-      history.push(`?contextId=${contextId}&iModelId=${iModelId}`);
-    }
-  }, [contextId, iModelId, isAuthorized]);
-
-  useEffect(() => {
-    if (isLoggingIn && isAuthorized) {
-      setIsLoggingIn(false);
-    }
-  }, [isAuthorized, isLoggingIn]);
-
-  const onLoginClick = async () => {
-    setIsLoggingIn(true);
-    await IModelApp.authorizationClient?.signIn();
-  };
-
-  const onLogoutClick = async () => {
-    setIsLoggingIn(false);
-    await IModelApp.authorizationClient?.signOut();
-    setIsAuthorized(false);
-  };
+  /** Grab the ContextId and IModelId from the URL parameters first, else the .ENV variables */
+  const urlParams = new URLSearchParams(window.location.search);
+  if (urlParams.has("contextId")) {
+    context.contextId = urlParams.get("contextId") as string;
```

Comment:
eh instead of directly mutating context, if you changed whats in context to also have a setState callback that might be better

Reply:
There are state variables now.